src/components/editTestCase/qdm/LeftPanel/ElementsTab/Elements/ElementsSection.tsx

```diff
@@ -122,6 +122,12 @@ const ElementsSection = (props: {
     setActiveTab(v);
   };
 
+  useEffect(() => {
+    if (selectedDataElement?.qdmCategory.length > 0) {
+      setActiveTab(selectedDataElement.qdmCategory);
+    }
+  }, [selectedDataElement]);
+
```

Comment:
To prevent extraneous rerenders and unforeseen side effects, could we instead make the change to the activeTab occur during the onView function? 
`setActiveTab` call could go on 199 to prevent the use of a useEffect

I believe React does multiple set updates in events so it would be similar to a single update in this way.

Reply:
shit, I think I have pushed the wrong changes this useEffect method was something which I initially used later on I have moved it to DataElementsTable itself, I think while I was cleaning the code I just removed the optimized code. thanks for pointing out Matt